File: marcenaria-java-activej/src/main/java/github/gmess/models/EstruturaValor.java
package github.gmess.models;

import github.gmess.models.geometrias.Geometria;

/**
 * EstruturaValor
 */
public record EstruturaValor(
  String estrutura,
  String geometria,
  String valor
) {

  public static EstruturaValor from(Geometria geometria, String valor) {
    return new EstruturaValor(
      geometria.getEstrutura(), 
      geometria.getGeometria(), 
      valor);
  }
}

File: MarcenariaDotNet/Dtos/EstruturaValor.cs
using MarcenariaDotNet.Dtos.Geometrias;

namespace MarcenariaDotNet.Dtos;

struct EstruturaValor
{
  public string Estrutura { get; init; }
  public string Geometria { get; init; }
  public double Valor { get; init; }

  public static EstruturaValor From(Geometria geometria, string material) {
    return new EstruturaValor {
      Estrutura = geometria.GetEstrutura(),
      Geometria = geometria.GetGeometria(),
      Valor = Material.CalculaPreco(geometria.GetArea(), material)
    };
  }
}
